src/CommonITILObject.php

```diff
@@ -8617,23 +8617,26 @@ protected function updateActors(bool $disable_notifications = false)
                     // Empty values, probably provided by static::getDefaultValues()
                     continue;
                 }
-
                 $found = false;
                 foreach ($existings as $existing) {
                     if (
                         $actor['itemtype'] === User::class
                         && $actor['items_id'] == 0
-                        && $actor['itemtype'] == $existing['itemtype']
+                        && $existing['itemtype'] === User::class
+                        && $existing['items_id'] == 0
                     ) {
-                        // "email" actor found
-                        if ($actor['alternative_email'] == $existing['alternative_email']) {
+                        // This is an existing anonymous actor
+                        if ($actor['alternative_email'] === $existing['alternative_email']) {
+                            // The anonymous actor matches an existing one, update it
+                            $updated[] = $actor + ['id' => $existing['id']];
                             $found = true;
                             break;
+                        } else {
+                            // "email" actor found but emails do not match
+                            // Do not check for modifications on "email" actors (they should be deleted then re-added on email change)
+                            continue;
                         }
-                        // Do not check for modifications on "email" actors (they should be deleted then re-added on email change)
-                        continue;
                     }
-
                     if ($actor['itemtype'] != $existing['itemtype'] || $actor['items_id'] != $existing['items_id']) {
```

Comment:
I'm not sure it was necessary to modify so many lines, only the addition of the line `$updated[] = ...` impact your test.

Reply:
The code has been retablished by keeping `$updated[]` line. https://github.com/glpi-project/glpi/pull/16070/commits/03f6105c12ac4734a1ba5b0afc91d25565dfaf58